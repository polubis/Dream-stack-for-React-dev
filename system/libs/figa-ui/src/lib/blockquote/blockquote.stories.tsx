import type { Story, Meta } from '@storybook/react';

import { Blockquote } from './blockquote';
import { Box } from '../box';
import { FONT_VARIANTS } from '../font';

export default {
  component: Blockquote,
  title: 'Blockquote',
} as Meta;

const Template: Story = () => {
  return (
    <Box spacing={FONT_VARIANTS.map(() => 150)}>
      {FONT_VARIANTS.map((variant) => (
        <Blockquote key={variant} variant={variant} children={variant} />
      ))}
    </Box>
  );
};

export const Default = Template.bind({});
Default.args = {};
