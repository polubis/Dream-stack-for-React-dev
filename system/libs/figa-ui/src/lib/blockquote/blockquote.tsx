import { Font } from '../font';
import type { BlockquoteProps } from './defs';

import c from 'classnames';

const Blockquote = ({ children, className, variant }: BlockquoteProps) => {
  return (
    <blockquote className={c('blockquote', className, variant)}>
      <Font variant={variant}>{children}</Font>
    </blockquote>
  );
};

export { Blockquote };
