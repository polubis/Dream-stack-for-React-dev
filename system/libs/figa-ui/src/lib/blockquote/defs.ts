import type { ReactNode } from 'react';
import type { FontVariant } from '../font';

type BlockquoteProps  ={
  className?: string;
  variant: FontVariant;
  children: ReactNode;
}

export type { BlockquoteProps };
