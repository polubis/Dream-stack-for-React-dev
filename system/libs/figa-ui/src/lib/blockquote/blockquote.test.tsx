import { render } from '@testing-library/react';
import { Blockquote } from './blockquote';

describe('Blockquote can be used when', () => {
  it('[FRAGILE] assigns classes', () => {
    const { asFragment } = render(
      <Blockquote variant="b2" className="my-class">
        My text
      </Blockquote>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
