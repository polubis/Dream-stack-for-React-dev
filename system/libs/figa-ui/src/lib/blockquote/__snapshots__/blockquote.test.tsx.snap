// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote can be used when [FRAGILE] assigns classes 1`] = `
<DocumentFragment>
  <blockquote
    class="blockquote my-class b2"
  >
    <p
      class="font b2 default"
    >
      My text
    </p>
  </blockquote>
</DocumentFragment>
`;
